perun-core/src/main/java/cz/metacentrum/perun/core/provisioning/GroupsHashedDataGenerator.java

```diff
@@ -87,7 +89,9 @@ public HashedGenData generateData() {
 			} else {
 				membersToEvaluate = sess.getPerunBl().getFacilitiesManagerBl().getAllowedMembers(sess, facility, service);
 			}
-			membersWithConsent.addAll(sess.getPerunBl().getConsentsManagerBl().evaluateConsents(sess, service, facility, membersToEvaluate));
+			if (consentEval) {
```

Comment:
Join into one if statement on line 85. We want to check if we force consents globally AND for that specific service